clang/include/clang/CIR/Dialect/IR/CIRAttrs.td

```diff
@@ -122,6 +122,37 @@ def CIR_OptInfoAttr : CIR_Attr<"OptInfo", "opt_info"> {
   let genVerifyDecl = 1;
 }
 
+//===----------------------------------------------------------------------===//
+// TypeSizeAttr
+//===----------------------------------------------------------------------===//
+
+def CIR_TypeSizesAttr : CIR_Attr<"TypeSizes", "type_sizes"> {
+  let summary = "the size of types in bits";
+  let description = [{
+    Attached to a module, to share information between CIR generation and
+    later passes.
+
+    Examples:
+
+    ```mlir
+    // sizeof(int) == 4, sizeof(size_t) == 8
+    module attributes {
+      cir.type_sizes = #cir.type_sizes<
+        int = 32,
+        size_t = 64
+      >
+    } {}
+    ```
+  }];
+
+  let parameters = (ins "unsigned":$intSize, "unsigned":$sizeTypeSize);
```

Comment:
Please add `extraClassDeclaration` with static methods that return the full type, example: `static mlir::Type getPtrDiffTy() { ... }`